python/lib/sift_py/data_import/hdf5.py

```
@@ -48,7 +48,7 @@ def upload(
         path: Union[str, Path],
         hdf5_config: Hdf5Config,
         show_progress: bool = True,
-    ) -> List[DataImportService]:
+    ) -> Optional[DataImportService]:
```

Why is this Optional?

---

I went back and forth on this one. We could originally return a blank list, which isn't possible to do with DataImportService, so we would need to return None instead. That could happen if the user provides a config/file that ultimately has no data in it to be processed.

Alternatively, we could raise an Exception if we didn't actually upload any files. If we keep it as-is, I'll add better documentation as to what None means

---

Made documentation more clear
https://github.com/sift-stack/sift/pull/261/commits/abcfe198cd20cd17e531ae72a9831cd78cdbda36

---

Done
https://github.com/sift-stack/sift/pull/261/commits/74d3189addfe481c544f43d9980d84a2212cb53f
Mypy corrections
https://github.com/sift-stack/sift/pull/261/commits/f1b90b1c88e8722d561894e2c26dc32a5cccb723